contrib/network/connection_balance/random/source/connection_balancer_impl.cc

```diff
@@ -0,0 +1,36 @@
+#include "contrib/network/connection_balance/random/source/connection_balancer_impl.h"
+
+#include <bits/types/FILE.h>
+#include <sys/eventfd.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <cstdlib>
+#include <memory>
+
+namespace Envoy {
+namespace Extensions {
+namespace Random {
+
+void RandomConnectionBalancerImpl::registerHandler(
+    Envoy::Network::BalancedConnectionHandler& handler) {
+  absl::MutexLock lock(&lock_);
+  handlers_.push_back(&handler);
+}
+
+void RandomConnectionBalancerImpl::unregisterHandler(
+    Envoy::Network::BalancedConnectionHandler& handler) {
+  absl::MutexLock lock(&lock_);
+  handlers_.erase(std::find(handlers_.begin(), handlers_.end(), &handler));
+}
+
+Envoy::Network::BalancedConnectionHandler&
+RandomConnectionBalancerImpl::pickTargetHandler(Envoy::Network::BalancedConnectionHandler&) {
+  absl::MutexLock lock(&lock_);
```

Comment:
I don't think we need the lock here. and the handlers list won't be changed after the Envoy startup. Otherwise, I don't think it can provide better performance than the exact balancer.

Reply:
Delete it.